helm/alert_payload_templates/ecs.json

```diff
@@ -0,0 +1 @@
+{"orchestrator":{"type":"kubernetes","cluster":{"name":"{{ cluster }}"},"namespace":"{{ namespace }}","resource":{"type":"Pod"}},"container":{"image":{"name":"{{ images }}"}},"agent":{"id":"{{ connaisseur_pod_id }}","type":"connaisseur"},"event":{"created":"{{ timestamp }}","id":"{{ request_id }}","severity":"{{ priority }}","category":"intrusion_detection","type":"denied","action":"{{ alert_message }}"},"ecs":{"version":"1.12.0"}}
```

Comment:
the only thing that seems problematic to me is that the ECS version is pinned in the last entry. That would make this seem ECS version specific. Do you see a way to avoid this?
Maybe @annekebr also has an idea.

Reply:
Does ECS require the `"ecs":{"version":<version>}}` object and does it reject the paylaod if it's not present?

If so, I would suggest either 
a) making the version configurable in the `helm/values.yaml` assuming that the schema of accepted JSON payloads did not/will not change too often and adding a comment with the ECS versions that are known to be compatible right now
b) adjust the docs to reflect that this template is for this specific version only and if someone wants to use another ECS version, they can use the template as template and create a custom template changing the version and potentially other fields in the JSON payload, if the used version requires a different format. 

What do you think?

Reply:
Hello @annekebr !
First of all, the `ecs.version` is required, and this is documented from elastic [1] (check the `core` level).

I believe that `a)` is a bit dirty, in the sense that we are not providing any other version, and we'll need to maintain ECS versions along the Helm chart.

I prefer `b)` and we could even rename the file to `ecs-1-12-0.json` to be very explicit.

----

[1] : https://www.elastic.co/guide/en/ecs/1.12/ecs-ecs.html#field-ecs-version

Reply:
Also, the `orchestrator` field is not available before ECS `1.10`.

Reply:
b) is perfectly fine and renaming the file sounds like a good idea :-) Please adapt the docs accordingly and add a short sentence what one would need to change to use another version (create custom template with adapted version field plus potential changes in the payload compatible with the version). 

Thanks a lot for your input! :-)

Reply:
Added some info on how to use ECS:
https://github.com/sse-secure-systems/connaisseur/pull/427/files#diff-dfedb6f792e2c252547cfc7fba0a5a3301668aaeb06f5b3a6332bde593990578R38

Transforming to different ECS is quite long scope for the doc. As there are so so many fields that could potentially be relevant. I suggest that, as ECS before `1.10.0` is unusable (as `orchestrator` map does not exist), we can wait for PRs for different ECS versions ( > `1.12.0`).